src/cmap/connection_pool.ts

```diff
@@ -494,13 +520,16 @@ function processWaitQueue(pool: any) {
       }
 
       if (err) {
-        pool.emit('connectionCheckOutFailed', new ConnectionCheckOutFailedEvent(pool, err));
-      } else {
+        pool.emit(
+          'connectionCheckOutFailed',
+          new ConnectionCheckOutFailedEvent(pool, err as MongoError)
```

Comment:
`ConnectionCheckOutFailedEvent` should accept whatever the union for our `ErrorLike` becomes (or whatever its called), so we can avoid casting here